### PR TITLE
feat: modify some Dart codes and fill in the empty Dart code blocks in docs

### PR DESCRIPTION
**Pull request by gvenusleo**
If this PR is related to coding or code translation, please fill out the checklist and paste the console outputs to the PR.

- [x] I've tested the code and ensured the outputs are the same as the outputs of reference codes.
- [x] I've checked the codes (formatting, comments, indentation, file header, etc) carefully.
- [x] The code does not rely on a particular environment or IDE and can be executed on a standard system (Win, macOS, Ubuntu).

### CODE REVIEW

**Review by krahets — APPROVED** [review]
Awesome! Thanks for the PR!

**Review by krahets — APPROVED** [review]
Awesome! Thanks for the PR!

**Comment by krahets** [comment]
Would you like to be the reviewer of Dart code for this book? 

- All the PRs related to Dart will be assigned to you for review.
- Your GitHub id will be listed in the preface section of the book together with other reviewers.

And if you want to trace the latest updates of the book, please add my WeChat krahets-jyd.

File: codes/dart/chapter_array_and_linkedlist/linked_list.dart [hunk 1]
```diff
@@ -45,9 +46,9 @@ int find(ListNode? head, int target) {
 }
 
 /* Driver Code */
-int main() {
+void main() {
```
Comment (krahets) [inline codes/dart/chapter_array_and_linkedlist/linked_list.dart:49]: two_sum.dart also use int main(), please update it.

**Comment by gvenusleo** [comment]
I would love to be the reviewer of Dart code! I've applied to add your WeChat. 

Also, I just submitted a PR to modify the `two_sum.dart` .

**Pull request merged**